test/expected/alter.out

```diff
@@ -194,3 +194,172 @@ SELECT * FROM _timescaledb_catalog.chunk WHERE id = 2;
+ _hyper_3_7_chunk | tablespace1
+ _hyper_3_8_chunk | tablespace1
+ _hyper_3_9_chunk | tablespace1
+ hyper_in_space   | tablespace1
+(6 rows)
+
+-- should not fail (unlike attach_tablespace)
+ALTER TABLE hyper_in_space SET TABLESPACE tablespace1;
+NOTICE:  tablespace "tablespace1" is already attached to hypertable "hyper_in_space", skipping
+\set ON_ERROR_STOP 0
+-- not an empty tablespace
+DROP TABLESPACE tablespace1;
+ERROR:  tablespace "tablespace1" is still attached to 1 hypertables
+\set ON_ERROR_STOP 1
+SELECT drop_chunks(20, 'hyper_in_space');
+ drop_chunks 
+-------------
+ 
+(1 row)
+
+SELECT tablename, tablespace FROM pg_tables WHERE tablespace = 'tablespace1' ORDER BY tablename;
+   tablename    | tablespace  
+----------------+-------------
+ hyper_in_space | tablespace1
+(1 row)
+
+\set ON_ERROR_STOP 0
+-- should not be able to drop tablespace if a hypertable depends on it
+-- even when there are no chunks
+DROP TABLESPACE tablespace1;
+ERROR:  tablespace "tablespace1" is still attached to 1 hypertables
+\set ON_ERROR_STOP 1
+DROP TABLE hyper_in_space;
+CREATE TABLE hyper_in_space(time bigint, temp float, device int) TABLESPACE tablespace1;
+SELECT create_hypertable('hyper_in_space', 'time', 'device', 4, chunk_time_interval=>1);
+NOTICE:  adding not-null constraint to column "time"
+ create_hypertable 
+-------------------
+ 
+(1 row)
+
+INSERT INTO hyper_in_space(time, temp, device) VALUES (1, 20, 1);
+INSERT INTO hyper_in_space(time, temp, device) VALUES (3, 21, 2);
+INSERT INTO hyper_in_space(time, temp, device) VALUES (5, 23, 1);
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename ~ '_hyper_\d+_\d+_chunk' ORDER BY tablename;
+     tablename     | tablespace  
+-------------------+-------------
+ _hyper_4_10_chunk | tablespace1
+ _hyper_4_11_chunk | tablespace1
+ _hyper_4_12_chunk | tablespace1
+ hyper_in_space    | tablespace1
+(4 rows)
+
+SELECT attach_tablespace('tablespace2', 'hyper_in_space');
+ attach_tablespace 
+-------------------
+ 
+(1 row)
+
+\set ON_ERROR_STOP 0
+-- should fail as >1 tablespaces are attached
+ALTER TABLE hyper_in_space SET TABLESPACE tablespace1;
+ERROR:  cannot ALTER TABLE SET TABLESPACE when multiple tablespaces are attached to hypertable hyper_in_space
+\set ON_ERROR_STOP 1
+SELECT detach_tablespace('tablespace2', 'hyper_in_space');
+ detach_tablespace 
+-------------------
+                 1
+(1 row)
+
+ALTER TABLE hyper_in_space SET TABLESPACE tablespace2;
+SELECT tablename, tablespace FROM pg_tables
+WHERE tablename = 'hyper_in_space' OR tablename ~ '_hyper_\d+_\d+_chunk' ORDER BY tablename;
+     tablename     | tablespace  
+-------------------+-------------
+ _hyper_4_10_chunk | tablespace2
+ _hyper_4_11_chunk | tablespace2
+ _hyper_4_12_chunk | tablespace2
+ hyper_in_space    | tablespace2
+(4 rows)
+
+-- attach tb1 <-> ALTER SET tb2 <-> detach tb2 should work
+SELECT detach_tablespace('tablespace2', 'hyper_in_space');
```

Comment:
what's the behavior of new chunks in this case? Which tablespace do they go into?

Reply:
Should be the same as if you did not attach or set tablespace. So put everything in whatever 'default_tablespace' points to.
Will add tests.